homeassistant/components/totalconnect/config_flow.py

```diff
@@ -36,22 +40,21 @@ async def async_step_user(self, user_input=None):
             await self.async_set_unique_id(username)
             self._abort_if_unique_id_configured()
 
-            client = await self.hass.async_add_executor_job(
-                TotalConnectClient, username, password, None
-            )
+            try:
+                client = await self.hass.async_add_executor_job(
+                    TotalConnectClient, username, password, None
+                )
+            except AuthenticationError:
+                errors["base"] = "invalid_auth"
+                return self.async_show_form(
+                    step_id="user", data_schema=data_schema, errors=errors
+                )
 
-            if client.is_logged_in():
-                # username/password valid so show user locations
-                self.username = username
-                self.password = password
-                self.client = client
-                return await self.async_step_locations()
-            # authentication failed / invalid
-            errors["base"] = "invalid_auth"
-
-        data_schema = vol.Schema(
-            {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
-        )
+            # username/password valid so show user locations
```

Comment:
Optional suggestion: You could consider the success case in an `else:` here and let the  error case fall through to the form below like it used to.

Reply:
The great `try ... except ... else` structure. I think it should be ported to more languages.

Reply:
done, thanks for the suggestion